etcdctl/ctlv3/command/watch_command.go

```diff
@@ -92,30 +99,55 @@ func watchInteractiveFunc(cmd *cobra.Command, args []string) {
 			continue
 		}
 
-		flagset := NewWatchCommand().Flags()
-		err = flagset.Parse(args[1:])
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "Invalid command %s (%v)\n", l, err)
-			continue
-		}
-		ch, err := getWatchChan(c, flagset.Args())
+		ch, execArgs, err := getWatchChan(c, args, true)
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "Invalid command %s (%v)\n", l, err)
 			continue
 		}
-		go printWatchCh(ch)
+		go printWatchCh(c, ch, execArgs)
 	}
 }
 
-func getWatchChan(c *clientv3.Client, args []string) (clientv3.WatchChan, error) {
-	if len(args) < 1 || len(args) > 2 {
-		return nil, fmt.Errorf("bad number of arguments")
+func getWatchChan(c *clientv3.Client, args []string, interactive bool) (clientv3.WatchChan, []string, error) {
+	if interactive {
+		args = args[1:] // args[0]=="watch"
+	}
+	if len(args) < 1 {
+		return nil, nil, errBadArgsNum
+	}
+
+	var execArgs []string
+	var err error
+	if interactive {
+		execArgs, err = findExecArgs(args)
+	} else {
+		// cobra strips out "--"; use raw os.Args
+		idx := 0
+		for idx = range os.Args {
+			if os.Args[idx] == "watch" {
+				break
+			}
+		}
+		osArgs := os.Args[idx+1:]
+		execArgs, err = findExecArgs(osArgs)
```

Comment:
Avoid repeating the `findExecArgs` call by restructuring this as:

```
if !interactive {
  ...
  args := osArgs
}
execArgs, err = findExecArgs(args)
```
?

Also, should findExecArgs return both the args before `--` as well as the args after `--` so that the subsequent code can deal with them as cleanly separated lists? I'm concerned that checks like `len(args) == 2 ` below might not make sense if we don't cleanly separate the two args lists.